src/scale_limits_analyzer.rs

```diff
@@ -0,0 +1,111 @@
+
+use bluejay_parser::error::{Annotation, Error as BluejayError};
+use bluejay_core::definition::ObjectTypeDefinition;
+use bluejay_core::definition::SchemaDefinition as CoreSchemaDefinition;
+use bluejay_core::AsIter;
+use bluejay_core::Directive;
+use bluejay_core::Value;
+use bluejay_core::{definition::HasDirectives, ValueReference};
+use bluejay_parser::{
+    ast::{
+        definition::FieldDefinition,
+        definition::{DefaultContext, DefinitionDocument, SchemaDefinition},
+        executable::ExecutableDocument,
+    },
+    Error,
+};
+
+pub type ScaleLimitsAnalyzer<'a> = bluejay_validator::executable::operation::Orchestrator<
+    'a,
+    ExecutableDocument<'a>,
+    SchemaDefinition<'a>,
+    serde_json::Map<String, serde_json::Value>,
+    ScaleLimits,
+>;
+
+pub struct ScaleLimits {
+  max_rate: f64,
+}
+
+impl
+    bluejay_validator::executable::operation::Visitor<
+        '_,
+        ExecutableDocument<'_>,
+        SchemaDefinition<'_>,
+        serde_json::Map<String, serde_json::Value>,
+    > for ScaleLimits
+{
+    fn new(
+        operation_definition: &'_ <ExecutableDocument as bluejay_core::executable::ExecutableDocument>::OperationDefinition,
+        schema_definition: &'_ SchemaDefinition,
+        variable_values: &'_ serde_json::Map<String, serde_json::Value>,
+        cache: &'_ bluejay_validator::executable::Cache<'_, ExecutableDocument, SchemaDefinition>,
+    ) -> Self {
+        Self { max_rate: 0.0 }
+    }
+
+    fn visit_field(
+        &mut self,
+        field: &'_ <ExecutableDocument<'_> as bluejay_core::executable::ExecutableDocument>::Field,
+        field_definition: &'_ <SchemaDefinition as CoreSchemaDefinition>::FieldDefinition,
+        scoped_type: bluejay_core::definition::TypeDefinitionReference<
+            '_,
+            <SchemaDefinition<'_> as CoreSchemaDefinition>::TypeDefinition,
+        >,
+        included: bool,
+    ) {
+        field_definition.directives().and_then(|directives| {
+            directives
+                .iter()
+                .find(|directive| directive.name() == "scaleLimits")
+                .and_then(|directive| {
+                    directive.arguments()
+                })
+                .and_then(|arguments| {
+                    arguments
+                        .iter()
+                        .find(|argument| argument.name() == "rate")
+                        .and_then(|argument| {
+                            if let ValueReference::Float(rate) = argument.value().as_ref() {
+                                self.update_max_rate(rate);
```

Comment:
We don't want to use just the max rate for all fields. Each field with `scaleLimits` may have a different rate. We should be following the same algorithm as what @jacobsteves implemented in Ruby.